core/src/main/java/hudson/logging/LogRecorder.java

```diff
@@ -148,7 +148,7 @@ public AutoCompletionCandidates doAutoCompleteLoggerName(@QueryParameter String
     }
 
     @Restricted(NoExternalUse.class)
-    transient /*almost final*/ RingBufferLogHandler handler = new RingBufferLogHandler() {
+    transient /*almost final*/ RingBufferLogHandler handler = new RingBufferLogHandler(RingBufferLogHandler.DEFAULT_RING_BUFFER_SIZE) {
```

Comment:
You can keep the previous line, the default size of the non-arg constructor is already `RingBufferLogHandler.DEFAULT_RING_BUFFER_SIZE`